docs/anonymity-configuration.rst

```diff
@@ -0,0 +1,521 @@
+with a publicly traceable TCP/IP server.
+
+**XXX is there a document maintained by Tor developers which substantiates or refutes this belief?
+If so we need to link to it. If not, then maybe we should explain more here why we think this?**
+
+Performance
+-----------
+
+A client connecting to a publicly traceable Tahoe-LAFS server through Tor
+incurs substantially higher latency and sometimes worse throughput than the
+same client connecting to the same server over a normal traceable TCP/IP
+connection. When the server is on a Tor Hidden Service, it incurs even more
+latency, and possibly even worse throughput.
+
+Connecting to Tahoe-LAFS servers which are I2P servers incurs higher latency
+and worse throughput too.
+
+Positive and negative effects on other Tor users
+------------------------------------------------
+
+Sending your Tahoe-LAFS traffic over Tor adds cover traffic for other
+Tor users who are also transmitting bulk data. So that is good for
+them -- increasing their anonymity.
+
+However, it makes the performance of other Tor users' interactive
+sessions -- e.g. ssh sessions -- much worse. This is because Tor
+doesn't currently have any prioritization or quality-of-service
+features, so someone else's ssh keystrokes may have to wait in line
+while your bulk file contents get transmitted. The added delay might
+make other people's interactive sessions unusable.
+
+Both of these effects are doubled if you upload or download files to a
+Tor Hidden Service, as compared to if you upload or download files
+over Tor to a publicly traceable TCP/IP server.
+
+Positive and negative effects on other I2P users
+------------------------------------------------
+
+Sending your Tahoe-LAFS traffic over I2P adds cover traffic for other I2P users
+who are also transmitting data. So that is good for them -- increasing their
+anonymity. It will not directly impair the performance of other I2P users'
+interactive sessions, because the I2P network has several congestion control and
+quality-of-service features, such as prioritizing smaller packets.
+
+However, if many users are sending Tahoe-LAFS traffic over I2P, and do not have
+their I2P routers configured to participate in much traffic, then the I2P
+network as a whole will suffer degradation. Each Tahoe-LAFS router using I2P has
+their own anonymizing tunnels that their data is sent through. On average, one
+Tahoe-LAFS node requires 12 other I2P routers to participate in their tunnels.
+
+It is therefore important that your I2P router is sharing bandwidth with other
+routers, so that you can give back as you use I2P. This will never impair the
+performance of your Tahoe-LAFS node, because your I2P router will always
+prioritize your own traffic.
+
+
+
+Torsocks: the old way of configuring Tahoe-LAFS to use Tor
+==========================================================
+
+Before the native Tor integration for Tahoe-LAFS, users would use Torsocks.
```

Comment:
note that using the latest stable torsocks no longer works at all with tahoe-lafs. an old version of torsocks must be used. we do not recommend this and neither does the author of torsocks ;-p